addons/mrp/models/mrp_production.py

```diff
@@ -1104,6 +1104,7 @@ def action_confirm(self):
             if not production.move_raw_ids:
                 raise UserError(_("Add some materials to consume before marking this MO as to do."))
             production.move_raw_ids._adjust_procure_method()
+            (production.move_raw_ids | production.move_finished_ids).origin = production.origin or production.procurement_group_id.name
```

Comment:
I fix the same issue (I hope) in my PR of "Date refactor" but differently : https://github.com/odoo/odoo/pull/55379/files#diff-b62e2447dcdf46e87d838527e24dc866R736
Do you see any situation where my fix doesn't work or is bad with your features ? We keep only one fix or both ?
What do you thing ?

Reply:
When the MO is created from replenishment report, currently the origin of MO and post manufacturing picking is "replenishment report". To keep this, maybe we can keep your fix but change to
`'origin': production.origin or production.name`

Reply:
You are right, my bad, I think we can keep both (my PR is already in merged state).